interfaces/builtin/docker_support.go

```diff
@@ -185,156 +185,20 @@ pivot_root,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-co[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-con[^f]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-conf[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-confi[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/core/*/usr/lib/snapd/snap-confin[^e]** -> cri-containerd.apparmor.d,
-
-# branch for the /snap/snapd/... paths
-change_profile unsafe /snap/snap[^d]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/[^u]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/u[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/us[^r]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/[^l]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/l[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/li[^b]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/s[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/sn[^a]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/sna[^p]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snap[^d]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd[^/]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/[^s]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/s[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/sn[^a]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/sna[^p]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap[^-]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-[^c]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-c[^o]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-co[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-con[^f]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-conf[^i]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confi[^n]** -> cri-containerd.apparmor.d,
-change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confin[^e]** -> cri-containerd.apparmor.d,
+# below are auto-generated rules using GenerateAAREExclusionPatterns
+%s
```

Comment:
using %s feels fragile, my preference would be to use a placeholder that contains some hint of what would be produced or the arguments passed to the helper:

###EXCL{change_profile unsafe <>  -> docker-default,:/snap/snapd/*/usr/lib/snapd/snap-confine,/snap/core/*/usr/lib/snapd/snap-confine}###

so the strawman for the placeholder here would be: ###EXCL{$PREFIX<>$SUFFIX:$STRINGS}### but anything sensible could work

then you could have a helper that takes the string with the placeholders plus the arguments to generate things, and the helper computes both the placeholder from the arguments and the generated content, and does the replacement or errors if it doesn't find the placeholder at all